services/QuillLMS/app/services/gen_ai/base_cloner.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module GenAI
+  class BaseCloner < ApplicationService
+    attr_reader :record
+
+    def initialize(record)
```

Comment:
Take or leave, but you can get rid of a lot of boilerplate code, but still have required structure, if you change this to the below. It does require you to pass more info into some of the `.run` calls, so it's really up to you if you think it's better (there are pros and cons to either approach).
```ruby
attr_reader :record, :merge_attributes

def initialize(record, merge_attributes = {})
  @record = record
  @merge_attributes = merge_attributes
  
  raise unless valid_merge_attributes?
end

private def valid_merge_attributes? = merge_attributes.keys.all? {|k| k.in?(self.class::MERGE_ATTRIBUTE_KEYS)}
```